### PR TITLE
PROD-30134: Move AsyncAPI definition for "event create" event into the social_event module

### PR DESCRIPTION
**Pull request by tregismoreira**
## Description

We want the AsyncAPI definitions to be co-located with the modules that produces the events. 

The first event "event create" was pushed to Cable Car because it was pending an script to [automatically aggregate all definitions into a single file](https://github.com/goalgorilla/cablecar/pull/1813). This PR moves it to `social_event` module.

Here's the workflow run that generates the final documentation after aggregating all definitions:
https://github.com/goalgorilla/cablecar/actions/runs/10149822534

## Issue tracker
https://getopensocial.atlassian.net/browse/PROD-30134

## Theme issue tracker
N/A

## How to test
N/A

## Screenshots
N/A

## Release notes
N/A

## Change Record
N/A

## Translations
N/A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3991-feyszmc66jjzvqnzj1io7vf2adb7u46p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66a3a9d99d8a16fc684452cc

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3991-feyszmc66jjzvqnzj1io7vf2adb7u46p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66a3a9d99d8a16fc684452cc

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3991-feyszmc66jjzvqnzj1io7vf2adb7u46p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66a3a9d99d8a16fc684452cc

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3991-feyszmc66jjzvqnzj1io7vf2adb7u46p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66a3a9d99d8a16fc684452cc

**Review by ribel — APPROVED** [review]
Thanks @tregismoreira! This approach looks good to me.
@bramtenhove could you please check if it is ok to you?

File: modules/social_features/social_event/asyncapi.yml [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+                      description: The creation time of the event.
+                    updated:
+                      type: string
+                      format: date-time
+                      description: The last updated time of the event.
+                    status:
+                      type: string
+                      description: The status of the event.
+                    label:
+                      type: string
```
Comment (bramtenhove) [inline modules/social_features/social_event/asyncapi.yml:29]: This is currently a string, but in fact we can have but two options only. Is there a way to define that or is that not possible @tregismoreira?

Comment (bramtenhove) [reply modules/social_features/social_event/asyncapi.yml:29]: Wrong field. I meant `status`, sorry!

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3991-feyszmc66jjzvqnzj1io7vf2adb7u46p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66a3a9d99d8a16fc684452cc

File: modules/social_features/social_event/asyncapi.yml (hunk 1 above)
Comment (bramtenhove) [reply modules/social_features/social_event/asyncapi.yml:29]: In the final documentation it'll not be clear that there is only `published` and `unpublished`. Just that they will receive a `string`.

If there is a way to define the `published` and `unpublished` that would be best imho.

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3991-feyszmc66jjzvqnzj1io7vf2adb7u46p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66a3a9d99d8a16fc684452cc

File: modules/social_features/social_event/asyncapi.yml (hunk 1 above)
Comment (tregismoreira) [reply modules/social_features/social_event/asyncapi.yml:32]: I added a property `enum` with those two options.

File: modules/social_features/social_event/asyncapi.yml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                      description: The status of the event.
+                      enum:
+                        - published
+                        - unpublished
+                    label:
+                      type: string
+                      description: The label of the event.
+                    visibility:
+                      type: string
+                      description: The visibility of the event content.
```
Comment (bramtenhove) [inline modules/social_features/social_event/asyncapi.yml:36]: Shouldn't this be an `object` too?

Either it's **Public**:
```
"visibility": {
  "type": "public"
}
```

**Community**:
```
"visibility": {
  "type": "community"
}
```

**Groups**:
```
"visibility": {
  "type": "groups"
  "groups": ["UUIDv4", etc.]
}
```

**Roles**:
```
"visibility": {
  "type": "roles"
  "roles": ["string", etc.]
}
```

File: modules/social_features/social_event/asyncapi.yml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+                          description: The second address line of the event location.
+                    enrollment:
+                      type: object
+                      properties:
+                        enabled:
+                          type: boolean
+                          description: Indicates if enrollment is enabled.
+                        method:
+                          type: string
+                          description: The method of enrollment.
```
Comment (bramtenhove) [inline modules/social_features/social_event/asyncapi.yml:122]:
```suggestion
                        method:
                          type: string
                          description: The method of enrollment.
                          enum:
                          - open
                          - invite
                          - request
```

Comment (bramtenhove) [reply modules/social_features/social_event/asyncapi.yml:122]: I also added a suggestion for this one based on what you did above with `enum` @tregismoreira

File: modules/social_features/social_event/asyncapi.yml (hunk 2 above)
Comment (bramtenhove) [reply modules/social_features/social_event/asyncapi.yml:36]: I see we currently only display the visibility value here: https://github.com/goalgorilla/open_social/commit/686d9a1e9408ff451178dc647ea4803a5d81f816#diff-98f8f30821c959c87cf6d1df16922c1f498b440f60ed5960f38a3d06c7065961R72.

@ribel, @tregismoreira, based on the initial payload design this should be an object though. Perhaps we can pick it up and address it outside of the scope of this one?
Sorry, I missed this before 😞

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3991-feyszmc66jjzvqnzj1io7vf2adb7u46p.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/66a3a9d99d8a16fc684452cc

File: modules/social_features/social_event/asyncapi.yml (hunk 3 above)
Comment (tregismoreira) [reply modules/social_features/social_event/asyncapi.yml:122]: pushed :)

File: modules/social_features/social_event/asyncapi.yml (hunk 2 above)
Comment (tregismoreira) [reply modules/social_features/social_event/asyncapi.yml:36]: We agreed to add a followup ticket for that.

**Review by bramtenhove — APPROVED** [review]

**Pull request merged**